### PR TITLE
Document Poisson solvers and charged-system convergence

### LINKED ISSUE #5632: Add docs for Poisson solvers and regtests for energy convergence of charged systems w.r.t. vacuum size
A crucial topic that is not yet covered by documentation is the Poisson solvers for different periodicities, how exactly a system with nonzero net charge is handled (#4442), and how the energy of a system converges with respect to the size of vacuum (if any) for neutral and charged cases.

This was originally sent to #4972 but did not receive response there. Now I think that its scope can be expanded beyond documentation as a set of tracking regtests could possibly resolve issues like #4038.

### PR DESCRIPTION
**Pull request by DCM-Uni-Paderborn**
## Summary

- add a dedicated DFT electrostatics page covering periodicity settings and supported Poisson solvers
- clarify that the wavelet solver requires FFTSG-compatible grid dimensions, which the default `EXTENDED_FFT_LENGTHS .FALSE.` guarantees independently of the selected FFT library
- document the homogeneous-background convention for charged 3D cells and the absence of a vacuum-independent limit for uncompensated charged 1D/2D systems
- add four focused `ANALYTIC` 2D regression tests for neutral and charged He at 10 and 20 Å cell lengths

## Motivation

Electrostatic boundary conditions are part of the physical model, but the manual did not explain how `CELL%PERIODIC`, `POISSON%PERIODIC`, and `IMPLICIT%BOUNDARY_CONDITIONS` interact. This also made charged-system energy comparisons across cell sizes or solvers easy to misinterpret.

The new tests track the neutral plateau and the expected cell-size dependence of an uncompensated charged 2D system independently. They avoid using the `IMPLICIT` input from #4038 as a 2D reference because its default `IMPLICIT%BOUNDARY_CONDITIONS PERIODIC` selects a 3D-periodic problem.

Closes #5632.

## Validation

- CP2K precommit checks for all nine changed files
- official regression-test driver with two MPI ranks: 4/4 tests passed
- one- and two-rank direct runs reproduced the reference energies
- generated the full input reference and rendered the new Sphinx page; all new input and bibliography links resolve

### CODE REVIEW

File: docs/methods/dft/electrostatics/index.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+| `PERIODIC`  | 3D                            | Standard fully periodic reciprocal-space solution.                                                                                                                          |
+| `ANALYTIC`  | 0D, 1D, 2D                    | Analytic reciprocal-space Green's functions; convergence can be slow.                                                                                                       |
+| `MT`        | 0D, 2D                        | Martyna--Tuckerman image decoupling. The cell must be at least twice the extent of the complete electronic density. [](#Martyna1999)                                        |
+| `MULTIPOLE` | 0D                            | Fits the total charge with atom-centred Gaussians to decouple periodic images. [](#Bl%C3%B6chl1995)                                                                         |
+| `WAVELET`   | 0D, 2D, 3D                    | Requires `PREFERRED_FFT_LIBRARY FFTSG`; the density must decay to zero at nonperiodic cell faces. [](#Genovese2006) [](#Genovese2007)                                       |
```
Comment (fstein93) [inline docs/methods/dft/electrostatics/index.md]: The wavelet solver requires the grid size to be compatible with `FFTSG`, not the library to be `FFTSG`. Thus, the keyword `EXTENDED_FFT_LENGTHS .FALSE.` (default)

**Review by fstein93 — CHANGES_REQUESTED** [review]

**Review by fstein93 — APPROVED** [review]
